### PR TITLE
25765/letters tests cypress

### PR DESCRIPTION
**Pull request by CBonade**
## Description
This rewrites the existing Nightwatch test to run in Cypress, testing the letters download form.  @erikphansen tagged you in this one as I ran into an issue regarding `window.Cypress` being included in the conditional for loading the mock data.  Having this included here, created a disconnect between the mock data that Nightwatch uses compared to what Cypress ends up using.  The key difference here, that causes an issue, is that the first step of the letters form expects an address to exist. In the `mockContactInformation`, an address gets injected into the React state appropriately, however in the `vet360ContactInformation`, the address is undefined, which causes the test to blow up on phase 1.  I considered rewriting the test, to go to the profile and add an address as part of the test, however traveling to the profile with the `vet360ContactInformation` version of the state loaded, results in an error where no address can be added.  ~To make this test work, I've removed `window.Cypress` as part of this conditional, however I am not sure what this may impact aside from this one test.  Tagged you as my git history shows your name on this code as the last to touch it, but if this is not yours, maybe you can please point me in the right direction of who to run this by?  Appreciate your time in advance!~

After investigating this further, it appeared that doing this broke a different test that was specifically targeting an error message around missing information. Decided on a different approach here of a quick band-aid of creating a conditional global variable to base the data set off of, with a proper refactor to be addressed in a different ticket.

## Testing done

With window.Cypress removed, this runs successfully in Cypress' Chrome runner.

## Screenshots
image

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by Mottie — APPROVED** [review]
Thanks!

**Review by k80bowman — APPROVED** [review]

**Comment by pjhill** [comment]
Resolves [25765](https://github.com/department-of-veterans-affairs/va.gov-team/issues/25765)

File: src/platform/user/profile/vap-svc/util/local-vapsvc.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ export function isVAProfileServiceConfigured() {
     // using the existence of VetsGov.pollTimeout as an indicator that we are
     // running unit tests and therefore _do_ want the FE to make real API calls
     window.VetsGov.pollTimeout ||
-    window.Cypress ||
+    (window.Cypress && !window.specRequiresAddress) ||
```
Comment (erikphansen) [inline src/platform/user/profile/vap-svc/util/local-vapsvc.js:9]: @CBonade I believe that if you properly mocked the `GET user/` call in your Cypress test this addition to the conditional would not be required. Can you please try that and revert this change?

Comment (CBonade) [reply src/platform/user/profile/vap-svc/util/local-vapsvc.js:9]: https://github.com/department-of-veterans-affairs/vets-website/pull/17656

Thanks for the feedback here!  I was able to rework the user mock as you suggested and get this test passing with a different user structure.  New PR linked above that adds the new user structure and removes the wacky conditional.  Thanks again!
